### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by michchen**
Added a popup for Amazon's choice tag (AmazonsChoice.jsx). Got CircleCI to work, added tests

### CODE REVIEW

**Comment by avincenthill** [comment]
Better commit messages, would prefer to see all imperative and descriptive

File: server.js [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
+app.use('/', express.static(path.join(__dirname, '/client/dist')));
+// app.use('/', express.static(__dirname + '/client/dist'));
 
 
-app.get("/get", (req, res) => {
-  // req.query is the URL query string, and "id" is the product id i wish to fetch
-  db.getProduct(req.query.id, data => {
+app.get('/get', (req, res) => {
+  // req.query is the URL query string, and 'id' is the product id i wish to fetch
+  db.getProduct(req.query.id, (data) => {
```
Comment (avincenthill) [inline server.js]: very cool

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+          - v1-dependencies-{{ checksum "package.json" }}
+          # fallback to using the latest cache if no exact match is found
+          - v1-dependencies-
+
+      - run: yarn install --ignore-engines
```
Comment (nr431207) [inline .circleci/config.yml]: if youre using npm, and not yarn, change the run to npm

File: client/src/components/PopupHistogram.jsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ const {
 
 const PopupHistogram = (props) => {
   const percentageArr = Object.entries(randomDistribution()).sort((x, y) => x[0] < y[0]);
-  const { rating, numReviews } = props;
+  const { rating, numReviews, boxHeight, boxWidth } = props;
```
Comment (mattminwoolee) [inline client/src/components/PopupHistogram.jsx]: Why don't you just destructure the props into the parameters for your class?
Ex. PopupHistogram = ( { rating, numReviews, boxHeight, boxWidth } ) => { }

File: client/src/components/PopupMessage.jsx [hunk 4]
```diff
@@ -1,12 +1,25 @@
+import styles from '../styles/Popup.css';
+
 const React = require('react');
 
 const PopupMessage = (props) => {
-  const { message } = props;
+  const { content, clickCb } = props;
+  const { stylesPopupMessage, stylesPopupMessageX } = styles;
```
Comment (mattminwoolee) [inline client/src/components/PopupMessage.jsx]: Is there a reason for this style of destructuring?

File: client/src/components/PopupMessage.jsx [hunk 5]
```diff
@@ -1,12 +1,25 @@
...
+  (close popup) is called. The onKeyPress accessibility feature is required by ESLint */
   return (
     <div>
-      {message}
-      <div>x</div>
+      <div className={stylesPopupMessage}>{content}</div>
+      <div
+        className={stylesPopupMessageX}
+        onClick={() => clickCb()}
+        onKeyPress={e => ((e.which === 88 || e.which === 27) ? clickCb() : '')}
```
Comment (mattminwoolee) [inline client/src/components/PopupMessage.jsx]: Nice.

File: database/cats.js [hunk 6]
```diff
@@ -1,34 +1,34 @@
...
-"http://static.cdnbridge.com/resources/18/160536/picture/83/85898883.jpg",
-"http://www.pethealthnetwork.com/sites/default/files/kitten-care162282356.png",
-"http://hhforcats.org/wp-content/uploads/2017/12/TwistShout2_sml.jpg",
-"http://www.villagevets.com/uploads/images/Belle1.jpg",
-"http://cdn.earthporm.com/wp-content/uploads/2015/03/before-and-after-growing-up-cats-18__880.jpg",
-"http://techwithkids.com/images/Little-Kitten-lead.jpg",
-"http://lh5.ggpht.com/_QE_t15aYSqg/SrAEty_xaZI/AAAAAAAABt0/ZVwoZACK7ho/Huff-Heri200.jpg",
-"http://www.valleycatsrescue.org/wp-content/uploads/2014/07/tom-tom1-980x460.jpg",
-"http://raisinghappykittens.com/wp-content/uploads/2015/05/kitten-growth-chart-5-day-o.jpg"]
+exports.data = ['http://www.petwebsite.co.uk/media/k2/items/cache/0548677e6432786dd8df61eb3aaec139_L.jpg',
```
Comment (mattminwoolee) [inline database/cats.js]: ty for all your cats. They've been very friendly to me

File: client/src/components/Rating.jsx [hunk 7]
```diff
@@ -31,23 +31,25 @@ class Rating extends React.Component {
     const { hover } = this.state;
     let { rating, numReviews } = this.props;
 
-    if (rating !== undefined && !isNaN(rating)) {
+    let ratingClassName = '';
+    if (rating !== undefined && !Number.isNaN(rating)) {
```
Comment (nr431207) [inline client/src/components/Rating.jsx]: maybe refactor to =>    if (rating && Number(rating))

File: .circleci/config.yml (hunk 2 above)
Comment (michchen) [reply .circleci/config.yml]: i ended up switching to yarn!

**Pull request merged**